lib/rules/no-implicit-coercion.js

```diff
@@ -245,28 +247,50 @@ module.exports = {
          * @returns {void}
```

Comment:
[Docs] I can't comment on the line, but the existing JSDoc for `shouldFix` is now a little misleading, I think. Previously it only meant the boolean of whether the report should include fixer. Now this PR adds to that meaning to mean whether the change should occur in a fix or suggestion.

Maybe: _"Whether this report should fix the node rather than provide a suggestion"_? Something like that? Thoughts?

Reply:
I think that's fine 👍🏻